src/test/java/teammates/test/BaseTestCaseWithLocalPostgresSqlDatabaseAccess.java

```diff
@@ -0,0 +1,48 @@
+package teammates.test;
+
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+import org.testcontainers.containers.JdbcDatabaseContainer;
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import teammates.common.datatransfer.sqlattributes.CourseAttributes;
+import teammates.common.util.HibernateUtil;
+import teammates.logic.api.LogicNewExtension;
+
+import javax.sql.DataSource;
+
+public abstract class BaseTestCaseWithLocalPostgresSqlDatabaseAccess extends BaseTestCaseWithPostgresSqlDatabaseAccess {
+    private final LogicNewExtension logic = new LogicNewExtension();
+    protected static PostgreSQLContainer pgsql;
+
+    @BeforeClass
+    public static void startContainer() {
+        pgsql = new PostgreSQLContainer<>("postgres:14.4")
+                .withDatabaseName("teammates")
+                .withUsername("teammates")
+                .withPassword("teammates");
+        pgsql.withInitScript("sql/V1__Create_course_table.sql");
+        pgsql.start();
+        HibernateUtil.setSessionFactoryForTesting(pgsql.getUsername(), pgsql.getPassword(), pgsql.getJdbcUrl());
+    }
+
+    @AfterClass
+    public static void closeContainer() {
+        pgsql.close();
```

Comment:
Seems that there's no method being called here to change the session factory back to the local session factory after testing?

Reply:
We can only create the `sessionFactory` when there is a postgre docker container running. Therefore, if we want to reset it back to the local session factory, when we want to do testing, we would also need to start up the dev postgre container, which I feel it is not the best way to do it.